mvp-spec/operator-design.md

```diff
@@ -52,197 +52,199 @@ Limitations:
-            O ->> O: check cmp.com is allowed to read AND WRITE
-            O ->> O: VERIFY full data signature<br/>with cmp.com public key
-            note over O: cmp.com is authenticated<br/>and timestamp can be trusted
-            O ->> O: check timestamp is in timeframe
-            note over O: ok: this is a valid request
-            O ->> O: read 1P 🍪: ID & sign[id]
-            O ->> O: VERIFY sign[pref & id]<br/>with cmp.com public key
-            note over O: ok: these preferences are valid<br>and they are associated with the id
-            O ->> O: write 1P 🍪:<br/>preferences & sign[pref & id]
-            note over O: preferences have been signed by cmp<br/>ID has been signed by operatorO.prebidsso.com<br/>both can be trusted
-            O ->> O: clear data = (id + sign[id]<br>+ preferences + sign[pref & id]<br>+ timestamp<br>+ sender=operatorO.prebidsso.com)
-            O ->> O: SIGN (clear data + receiver=publisherP.com)<br/> with own private key
-            O -->> B: REDIRECT
-        deactivate O
-
-        activate B
-            B ->> P: GET /pageP.html?[...clear Data]&signature=xxx
-        deactivate B
-
-        activate P
-            P -->> B: display
-        deactivate P
-
-        activate B
-            note over B: sender == operatorO.prebidsso.com<br>key == operatorO.prebidsso.com public key
-            B ->> B: VERIFY full data signature<br/>with operatorO.prebidsso.com public key
-            note over B: operatorO.prebidsso.com is authenticated<br/>and timestamp can be trusted
-            B ->> B: check timestamp is in timeframe
-            note over B: ok: this is a valid request
-            B ->> B: VERIFY sign[id] with operatorO.prebidsso.com public key
-            B ->> B: VERIFY sign[pref & id] with cmp.com public key
-            B ->> B: write 1P 🍪:<br/>id & sign[id]<br>preferences & sign[pref & id]
-        deactivate B
-
-        B -->> U: 
-
-    deactivate U
-end
-
-rect rgba(255, 255, 255, .55)
-    note right of U: Time T2<br>"READ" scenario: user visits a "new" advertiser
-
-    U ->> B: visit www.advertiserA.com/pageA.html
-    activate U
-
-        activate B
-            B ->> A: GET /pageA.html
-        deactivate B
-
-        activate A
-        A ->> A: no 🍪 found
-        A ->> A: clear data = (timestamp<br>+ sender=advertiserA.com<br>+ returnUrl=advertiserA.com/pageA.html)
-        A ->> A: SIGN (clear data + receiver=operatorO.prebidsso.com)<br/> with own private key
-        A -->> B: REDIRECT
-        deactivate A
-
-        activate B
-        B ->> O: GET or POST /read?[...clear Data]&signature=xxx
-        deactivate B
-
-        activate O
-        note over O: sender == advertiserA.com<br>key == advertiserA.com public key
-        O ->> O: check advertiserA.com is allowed to read
-        O ->> O: VERIFY full data signature<br/>with advertiserA.com public key
-        note over O: advertiserA is authenticated<br/>and timestamp can be trusted
+    participant B as Browser
```

Comment:
I understand you removed the "User" participant to clean the diagram, but I'm a little concerned because the goal was to highlight two **independant** steps:
- the user accesses the page
- the user submits their preferences

It's important to understand they happen at different times (there can be 5 - 10 minutes between both) and can be seen as two "sequences".

This step seems mandatory to me:
```
U ->> B: submit preferences
```

Reply:
Also, I remember discussions with product owners that didn't really understand what were the _triggers_ (ie. what user actions).
I recommend putting back the U participant to both diagrams.